### PR TITLE
[WIP] Concurrent Indexing UI

### PR DESCRIPTION
**Pull request by priyashpatil**
## Description
- [x] Add id support on IndexingProgressUpdate
- [x] Enable publishing docs indexing updates
- [x] fixes docs force indexing ran before indexing 
- [x] add concurrent docs indexing
- [x] fixes style props passing through StyledTooltip
- [x] Update gui for concurrent Indexing Progress Updates
- [ ] Fix UI states for showing errors coming from concurrent operations

## Checklist

- [x] The base branch of this PR is `dev`, rather than `main`
- [x] The relevant docs, if any, have been updated or created

## Screenshots
https://github.com/user-attachments/assets/95e00834-eef7-49e1-a8a5-f2ce89baaeb1

### CODE REVIEW

**Comment by priyashpatil** [comment]
Getting SQLite error while indexing both code and docs.
```
console.ts:137 [Extension Host] SQLITE_READONLY: attempt to write a readonly database (at console.<anonymous> (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:144:47509))
```
steps to reproduce: delete index folder, trigger code indexing, trigger docs indexing. **concurrent docs indexing works fine.**

File: core/core.ts [hunk 1]
```diff
@@ -286,11 +288,17 @@ export class Core {
...
+      const indexingPromises = siteIndexingOptions.map(site =>
+        this.getEmbeddingsProviderAndIndexDoc(site, msg.data.reIndex)
+      );
 
-      this.ide.infoPopup("Docs indexing completed");
+      // Await for all promises to resolve
+      try {
+        await Promise.all(indexingPromises);
+      } catch (error: any) {
+        console.error(error.message);
```
Comment (spew) [inline core/core.ts]: This should probably continue to throw rather than silently fail (from the user perspective).

**Comment by Patrick-Erichsen** [comment]
@priyashpatil #1864 is about ready to be merged with should fix the sqlite error you experienced. The merge conflict for your work in DocsService should be easy to resolve since the PR doesn't modify the yield logic you've updated.

**Comment by priyashpatil** [comment]
> @priyashpatil #1864 is about ready to be merged with should fix the sqlite error you experienced. The merge conflict that will arise from your changes to DocsService in this PR should be easy to resolve since my branch doesn't modify the yield logic you've updated.

Noted

**Comment by priyashpatil** [comment]
I am closing this due to too many conflicts. Sending this into small PRs.

**Pull request closed without merging**